sklearn/linear_model/base.py

```diff
@@ -273,6 +273,11 @@ def _set_intercept(self, X_offset, y_offset, X_scale):
         """Set the intercept_
         """
         if self.fit_intercept:
+            if isinstance(self.coef_, np.ndarray):
```

Comment:
Yes, but that is not the issue right now. For example, this works:

```python
a = [1, 2, 3]
b = np.array([1, 2, 3])
a / b
array([ 1.,  1.,  1.])
```

The issue is that `X_offset` is being cast to `np.float64` in `_preprocess_data`, which should not happen..